tagless-final/src/main/java/com/github/jorgecastillo/kotlinandroid/tagless/algebras/ui/Presentation.kt

```diff
@@ -33,46 +36,44 @@ interface SuperHeroDetailView : SuperHeroesView {
 
 }
 
-@typeclass
-interface Presentation<F> : TC {
-
-    fun navigation(): Navigation<F>
-
-    fun heroesService(): HeroesUseCases<F>
-
-    fun ME(): MonadError<F, Throwable>
-
-    fun onHeroListItemClick(ctx: Context, heroId: String): HK<F, Unit> =
-            navigation().goToHeroDetailsPage(ctx, heroId)
-
-    fun displayErrors(view: SuperHeroesView, t: Throwable): HK<F, Unit> =
-            ME().pure(when (CharacterError.fromThrowable(t)) {
-                is CharacterError.NotFoundError -> view.showNotFoundError()
-                is CharacterError.UnknownServerError -> view.showGenericError()
-                is CharacterError.AuthenticationError -> view.showAuthenticationError()
-            })
+class Presentation<F> @Inject constructor(
+        val navigation: Navigation<F>,
+        val heroesService: HeroesUseCases<F>,
+        val monadSuspend: MonadSuspend<F>) {
+
+    fun onHeroListItemClick(ctx: Context, heroId: String): Kind<F, Unit> =
+            navigation.goToHeroDetailsPage(ctx, heroId)
+
+    fun displayErrors(view: SuperHeroesView, t: Throwable): Kind<F, Unit> =
+            monadSuspend {
+                when (CharacterError.fromThrowable(t)) {
+                    is CharacterError.NotFoundError -> view.showNotFoundError()
+                    is CharacterError.UnknownServerError -> view.showGenericError()
+                    is CharacterError.AuthenticationError -> view.showAuthenticationError()
+                }
+            }
 
-    fun drawSuperHeroes(view: SuperHeroesListView): HK<F, Unit> =
-            ME().binding {
-                val result = ME().handleError(heroesService().getHeroes(), { displayErrors(view, it); emptyList() }).bind()
-                ME().pure(view.drawHeroes(result.map {
+    fun drawSuperHeroes(view: SuperHeroesListView): Kind<F, Unit> =
+            monadSuspend.bindingCatch {
```

Comment:
You use a `bindingCatch` but you are handling errors manually with the monadSuspend instance, inside of the binding block. What's this catch being used for ?

If it's about UI rendering exceptions on the following lines, those must not be catched but let the program crash IMO, since not rendering anything on screen is **not** something the end user of the code / app should afford. 

I.e: If something fails on your network layer, it's ok to catch it and notify an error to the user, but if something fails on rendering code and you catch it, the user does not have a way to know what's happening, he'd just see a blank screen.